Sources/Core/JavaIR.swift

```diff
@@ -133,24 +171,34 @@ public struct JavaIR {
 
         func render() -> [String] {
             let implementsList = implements?.joined(separator: ", ") ?? ""
-            let implementsStmt = implementsList == "" ? "" : "implements \(implementsList)"
-
-            var propertiesStrings = [String]()
-            for propertyBatch in properties {
-                for property in propertyBatch {
-                    propertiesStrings.append(property.render())
-                }
-                propertiesStrings.append("") // Add an empty line between each batch
-            }
+            let implementsStmt = implementsList == "" ? "" : " implements \(implementsList)"
 
-            return annotations.map { "@\($0)" } + [
-                "\(modifiers.render()) class \(name) \(implementsStmt) {",
-                -->enums.flatMap { $0.render() },
-                -->propertiesStrings,
-                -->methods.flatMap { $0.render() },
-                -->innerClasses.flatMap { $0.render() },
-                "}",
+            let extendsStmt = extends.map { " extends \($0) " } ?? ""
+
+            var lines = annotations.map { "@\($0)" } + [
+                "\(modifiers.render()) class \(name)\(extendsStmt)\(implementsStmt) {",
             ]
+
+            if !enums.isEmpty {
+                lines.append("")
+                lines.append(-->enums.flatMap { $0.render() })
```

Comment:
should we combine these lines like we do in the `properties` and `methods` sections?

Reply:
Done.